packages/augur-core/source/contracts/reporting/Market.sol

```diff
@@ -256,21 +259,47 @@ contract Market is ITyped, Initializable, Ownable, IMarket {
         return _amount / feeDivisor;
     }
 
-    function recordMarketCreatorFees(uint256 _marketCreatorFees) public returns (bool) {
+    function recordMarketCreatorFees(uint256 _marketCreatorFees, address _affiliateAddress) public returns (bool) {
         require(augur.isKnownFeeSender(msg.sender));
+        if (_affiliateAddress != NULL_ADDRESS && affiliateFeeDivisor != 0) {
+            uint256 _affiliateFees = _marketCreatorFees / affiliateFeeDivisor;
+            affiliateFeesAttoEth[_affiliateAddress] = _affiliateFees;
+            _marketCreatorFees = _marketCreatorFees.sub(_affiliateFees);
+        }
         marketCreatorFeesAttoEth = marketCreatorFeesAttoEth.add(_marketCreatorFees);
+        if (isFinalized()) {
+            distributeMarketCreatorFees(_affiliateAddress);
+        }
     }
 
     function distributeValidityBondAndMarketCreatorFees() private returns (bool) {
         // If the market resolved to invalid the bond gets sent to the auction. Otherwise it gets returned to the market creator.
-        uint256 _bondAndFees = validityBondAttoEth.add(marketCreatorFeesAttoEth);
+        marketCreatorFeesAttoEth = validityBondAttoEth.add(marketCreatorFeesAttoEth);
+        return distributeMarketCreatorFees(NULL_ADDRESS);
+    }
+
+    function distributeMarketCreatorFees(address _affiliateAddress) private returns (bool) {
         if (!isInvalid()) {
-            cash.transfer(owner, _bondAndFees);
+            cash.transfer(owner, marketCreatorFeesAttoEth);
+            if (_affiliateAddress != NULL_ADDRESS) {
+                withdrawAffiliateFees(_affiliateAddress);
+            }
         } else {
             IAuction _auction = universe.getAuction();
-            cash.transfer(universe.getAuction(), _bondAndFees);
-            _auction.recordFees(_bondAndFees);
+            cash.transfer(universe.getAuction(), marketCreatorFeesAttoEth);
+            _auction.recordFees(marketCreatorFeesAttoEth);
+        }
+        marketCreatorFeesAttoEth = 0;
```

Comment:
Wouldn't it be safer to move this line before the if/else block?

Reply:
Since the cash contract isn't a third party contract and we're just doing an ERC20 transfer there's no opportunity here for re-entrancy through the cash transfer, though it would be nicer for auditing. I'll note to revisit doing this in the future since currently we're only under the market contract size limit by 29 bytes (but it will get reduced in an upcoming bit of work where we can add this in)

Reply:
Ok, sounds good!